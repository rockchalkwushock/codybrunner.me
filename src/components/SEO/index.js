import React from 'react'
import Helmet from 'react-helmet'
import { bool } from 'prop-types'

import { mergeStrings } from '../../lib/helpers'

/**
 * @fileOverview
 * SEO Component
 *
 * Accepts post & site meta data from GraphQL Query.
 * postSeo {Boolean} tells component whether or not to
 * use post or site meta data.
 *
 * TODO: Images/Favicons/Icons
 */

const SEO = ({ post, postSeo, site }) => {
  const keywords = postSeo
    ? mergeStrings(site.keywords, post.frontmatter.keywords)
    : site.keywords
  return (
    <Helmet>
      {/* General Tags */}
      <html
        itemType={
          postSeo ? 'http://schema.org/Article' : 'http://schema.org/WebSite'
        }
        lang={site.lang}
        prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#"
      />
      <meta name="author" content={site.author} />
      <meta
        name="description"
        content={postSeo ? post.frontmatter.description : site.description}
      />
      <meta name="keywords" content={keywords} />
      {process.env.NODE_ENV === 'production' ? (
        <base target="_blank" href={site.siteUrl} />
      ) : null}
      {/* Schema.org Tags */}
      <meta
        itemProp="name"
        content={postSeo ? post.frontmatter.title : site.title}
      />
      <meta
        itemProp="description"
        content={postSeo ? post.frontmatter.description : site.description}
      />
      <meta
        itemProp="datePublished"
        content={postSeo ? post.frontmatter.date : ''}
      />
      <meta
        itemProp="dateModified"
        content={postSeo ? post.frontmatter.date : ''}
      />
      <meta
        itemProp="wordCount"
        content={postSeo ? post.frontmatter.wordCount : ''}
      />
      <meta itemProp="keywords" content={keywords} />
      <meta itemProp="image" content="" />
      {/* OpenGraph Tags */}
      <meta
        property="og:title"
        content={postSeo ? post.frontmatter.title : site.title}
      />
      <meta
        property="og:description"
        content={postSeo ? post.frontmatter.description : site.description}
      />
      <meta property="og:image" content="" />
      <meta property="og:image:alt" content="" />
      <meta property="og:image:type" content="image/jpg" />
      <meta property="og:image:height" content="1256" />
      <meta property="og:image:width" content="2400" />
      <meta
        property="og:url"
        content={postSeo ? `${site.siteUrl}${post.fields.slug}` : site.siteUrl}
      />
      <meta property="og:site_name" content={site.title} />
      <meta property="og:locale" content={site.lang} />
      <meta property="og:type" content={postSeo ? 'article' : 'website'} />
      {/* Twitter Tags */}
      <meta name="twitter:card" content="summary" />
      <meta name="twitter:creator" content={site.twitter} />
      <meta
        name="twitter:title"
        content={postSeo ? post.frontmatter.title : site.title}
      />
      <meta
        name="twitter:description"
        content={postSeo ? post.frontmatter.description : site.description}
      />
      <meta name="twitter:image" content="" />
      <meta name="twitter:image:alt" content="" />
      <meta
        name="twitter:url"
        content={postSeo ? `${site.siteUrl}${post.fields.slug}` : site.siteUrl}
      />
      <title itemProp="name" lang={site.lang}>
        {postSeo ? `Cody's Blog - ${post.frontmatter.title}` : site.title}
      </title>
    </Helmet>
  )
}

SEO.propTypes = {
  postSeo: bool
}

export default SEO
