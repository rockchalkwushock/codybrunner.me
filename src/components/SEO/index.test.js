import React from 'react'
import { shallow } from 'enzyme'

import SEO from './'

const siteData = {
  author: 'Turd Ferguson',
  description: "Turd's Space",
  googleVerify: 'sjfasdfasdfsdf',
  keywords: "snl, that's a funny name",
  lang: 'sarcasm',
  siteUrl: 'turdspace.com',
  title: "That's a Funny Name",
  twitter: '@theRealTurd'
}

const postData = {
  fields: {
    slug: '/turdsspace/'
  },
  frontmatter: {
    date: '02-11-99',
    description: 'Yabba dabba do',
    draft: false,
    keywords: 'something borrowed, something earned',
    tags: ['snl', 'funny business', 'testing'],
    title: 'A funny article from a funny guy'
  },
  wordCount: {
    words: 567
  }
}

describe('Component: <SEO />', () => {
  let wrapper
  let post
  beforeEach(() => {
    wrapper = shallow(<SEO site={siteData} />)
    post = shallow(<SEO post={postData} postSeo site={siteData} />)
  })

  test('should render correctly as site data', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should render correctly as post data', () => {
    expect(post).toMatchSnapshot()
  })
  test('should show as itemType: Article if post', () => {
    expect(
      post
        .find('html')
        .prop('itemType')
        .substr(18)
    ).toEqual('Article')
  })
  test('should show as itemType: WebSite if not post', () => {
    expect(
      wrapper
        .find('html')
        .prop('itemType')
        .substr(18)
    ).toEqual('WebSite')
  })
})
