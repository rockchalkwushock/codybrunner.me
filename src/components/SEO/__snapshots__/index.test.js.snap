// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <SEO /> should render correctly as post data 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <html
    itemType="http://schema.org/Article"
    lang="sarcasm"
    prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#"
  />
  <meta
    content="Turd Ferguson"
    name="author"
  />
  <meta
    content="Yabba dabba do"
    name="description"
  />
  <meta
    content="snl, that's a funny name, something borrowed, something earned"
    name="keywords"
  />
  <meta
    content="A funny article from a funny guy"
    itemProp="name"
  />
  <meta
    content="Yabba dabba do"
    itemProp="description"
  />
  <meta
    content="02-11-99"
    itemProp="datePublished"
  />
  <meta
    content="02-11-99"
    itemProp="dateModified"
  />
  <meta
    content={567}
    itemProp="wordCount"
  />
  <meta
    content="snl, that's a funny name, something borrowed, something earned"
    itemProp="keywords"
  />
  <meta
    content={null}
    itemProp="image"
  />
  <meta
    content="A funny article from a funny guy"
    property="og:title"
  />
  <meta
    content="Yabba dabba do"
    property="og:description"
  />
  <meta
    content={null}
    property="og:image"
  />
  <meta
    content="Logo for codybrunner.rocks"
    property="og:image:alt"
  />
  <meta
    content="image/jpeg"
    property="og:image:type"
  />
  <meta
    content="300"
    property="og:image:height"
  />
  <meta
    content="400"
    property="og:image:width"
  />
  <meta
    content="turdspace.com/turdsspace/"
    property="og:url"
  />
  <meta
    content="That's a Funny Name"
    property="og:site_name"
  />
  <meta
    content="sarcasm"
    property="og:locale"
  />
  <meta
    content="article"
    property="og:type"
  />
  <meta
    content="summary"
    name="twitter:card"
  />
  <meta
    content="@theRealTurd"
    name="twitter:creator"
  />
  <meta
    content="A funny article from a funny guy"
    name="twitter:title"
  />
  <meta
    content="Yabba dabba do"
    name="twitter:description"
  />
  <meta
    content={null}
    name="twitter:image"
  />
  <meta
    content="Logo for codybrunner.rocks"
    name="twitter:image:alt"
  />
  <meta
    content="turdspace.com/turdsspace/"
    name="twitter:url"
  />
  <title
    itemProp="name"
    lang="sarcasm"
  >
    Cody's Blog - A funny article from a funny guy
  </title>
</HelmetWrapper>
`;

exports[`Component: <SEO /> should render correctly as site data 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <html
    itemType="http://schema.org/WebSite"
    lang="sarcasm"
    prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#"
  />
  <meta
    content="Turd Ferguson"
    name="author"
  />
  <meta
    content="Turd's Space"
    name="description"
  />
  <meta
    content="snl, that's a funny name"
    name="keywords"
  />
  <meta
    content="That's a Funny Name"
    itemProp="name"
  />
  <meta
    content="Turd's Space"
    itemProp="description"
  />
  <meta
    content=""
    itemProp="datePublished"
  />
  <meta
    content=""
    itemProp="dateModified"
  />
  <meta
    content=""
    itemProp="wordCount"
  />
  <meta
    content="snl, that's a funny name"
    itemProp="keywords"
  />
  <meta
    content={null}
    itemProp="image"
  />
  <meta
    content="That's a Funny Name"
    property="og:title"
  />
  <meta
    content="Turd's Space"
    property="og:description"
  />
  <meta
    content={null}
    property="og:image"
  />
  <meta
    content="Logo for codybrunner.rocks"
    property="og:image:alt"
  />
  <meta
    content="image/jpeg"
    property="og:image:type"
  />
  <meta
    content="300"
    property="og:image:height"
  />
  <meta
    content="400"
    property="og:image:width"
  />
  <meta
    content="turdspace.com"
    property="og:url"
  />
  <meta
    content="That's a Funny Name"
    property="og:site_name"
  />
  <meta
    content="sarcasm"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="summary"
    name="twitter:card"
  />
  <meta
    content="@theRealTurd"
    name="twitter:creator"
  />
  <meta
    content="That's a Funny Name"
    name="twitter:title"
  />
  <meta
    content="Turd's Space"
    name="twitter:description"
  />
  <meta
    content={null}
    name="twitter:image"
  />
  <meta
    content="Logo for codybrunner.rocks"
    name="twitter:image:alt"
  />
  <meta
    content="turdspace.com"
    name="twitter:url"
  />
  <title
    itemProp="name"
    lang="sarcasm"
  >
    That's a Funny Name
  </title>
</HelmetWrapper>
`;
